gradle-plugin-testing/src/test/groovy/com/palantir/gradle/plugintesting/PluginTestingPluginIntegrationSpec.groovy

```diff
@@ -129,9 +128,43 @@ class PluginTestingPluginIntegrationSpec extends AbstractTestingPluginSpec {
         version << GradleTestVersions.gradleVersionsForTests
     }
 
+    def 'use the test driver Gradle version when a version is not set: #version'() {
```

Comment:
I think you also need a test that the file does what it says i.e. can you run the build under test on a single gradle version, but set it up that the config file runs two versions. That a test you've written runs against two versions.

There's nothing here in that class that does that. You've got deserialisation logic testing and that's it, there's nothing that indicates that you've wired things up correctly.